docs/ref/models/database-functions.txt

```diff
@@ -364,6 +381,81 @@ Usage example::
     >>> print(completed_same_year_count)
     1
 
+The specific ``Extract`` subclasses listed below are all fairly similar in
+function, except that those which extract time components can not be used on
+``DateField``\s.
+
+``DateTimeField`` examples::
+
+    >>> from datetime import datetime
+    >>> from django.utils import timezone
+    >>> from django.db.models.functions import (
+        ExtractYear, ExtractMonth, ExtractDay, ExtractWeekDay,
+        ExtractHour, ExtractMinute, ExtractSecond,
+    )
+    >>> start_2015 = datetime(2015, 6, 15, 23, 30, 1, tzinfo=timezone.utc)
+    >>> end_2015 = datetime(2015, 6, 16, 13, 11, 27, tzinfo=timezone.utc)
+    >>> Experiment.objects.create(
+    ...    start_time=start_2015, start_date=start_2015.date(),
+    ...    end_time=end_2015, end_date=end_2015.date())
+    >>> print(
+    ...    Experiment.objects.annotate(
+    ...        year=ExtractYear('start_time'),
+    ...        month=ExtractMonth('start_time'),
+    ...        day=ExtractDay('start_time'),
+    ...        weekday=ExtractWeekDay('start_time'),
+    ...        hour=ExtractHour('start_time'),
+    ...        minute=ExtractMinute('start_time'),
+    ...        second=ExtractSecond('start_time'))
+    ...    .values('year', 'month', 'day', 'weekday', 'hour', 'minute', 'second')
+    ...    .get(end_time__year=ExtractYear('start_time'))
+    ... )
+    {
+        'year': 2015, 'month': 6, 'day': 15, 'weekday': 2,
+        'hour': 23, 'minute': 30, 'second': 1
+    }
+
+    >>> # When a user has a different timezone activated then results will
```

Comment:
To break up the large code block, you could convert these comments into a regular paragraph.